packages/components/src/ui/item-group/item-group.tsx

```diff
@@ -7,26 +7,35 @@ import type { Ref } from 'react';
 /**
  * Internal dependencies
  */
-import type { PolymorphicComponentProps } from '../context';
-// eslint-disable-next-line no-duplicate-imports
-import { contextConnect } from '../context';
-import { useItemGroup } from './use-item-group';
-// eslint-disable-next-line no-duplicate-imports
-import type { Props } from './use-item-group';
+import {
+	contextConnect,
+	useContextSystem,
+	PolymorphicComponentProps,
+} from '../context';
 import { ItemGroupContext, useItemGroupContext } from './context';
-import { View } from '../../view';
+import { ItemGroupView } from './styles';
+import type { ItemGroupProps } from './types';
+
+const DEFAULT_PROPS = {
+	size: 'medium' as const,
+	isRounded: true,
+};
 
 function ItemGroup(
-	props: PolymorphicComponentProps< Props, 'div' >,
+	props: PolymorphicComponentProps< ItemGroupProps, 'div' >,
 	forwardedRef: Ref< any >
 ) {
-	const { bordered, separated, size: sizeProp, ...otherProps } = useItemGroup(
-		props
-	);
+	const {
+		isBordered = false,
+		isSeparated = false,
+		size: sizeProp,
+		role = 'list',
+		...otherProps
+	} = useContextSystem( props, 'ItemGroup' );
```

Comment:
Should we move these default prop values to the `DEFAULT_PROPS` object?

We could then rewrite this something like:

```suggestion
	const {
		isBordered,
		isSeparated,
		size: sizeProp,
		role,
		...otherProps
	} = useContextSystem( { ...DEFAULT_PROPS, props }, 'ItemGroup' );
```

and then remove spreading `DEFAULT_PROPS` directly on `<ItemGroupView />`